lua/autorun/client/cl_devtools.lua

```diff
@@ -1,144 +1,171 @@
-﻿-- This file is subject to copyright - contact swampservers@gmail.com for more information.
+﻿--[[
+    Limitations:
+        - If a path has lua/STRUCTHERE eg. "lua/autorun/lua/weapons/test.lua" then
+            It'll pass the Struct into it, not sure if it's a bug or a feature.
+        - A file within a sub directory of a struct folder like: "lua/weapons/testEnt/mysubdir/lua"
+            Won't have the SWEP/struct table passed to it.
+
+]]--
+
+-- This file is subject to copyright - contact swampservers@gmail.com for more information.
 -- INSTALL: CINEMA
 SWAMP_DEV = SWAMP_DEV or {}
-SWAMP_DEV.isDev = SWAMP_DEV.isDev
+
+SWAMP_DEV.allowed = {
+    ["SteamID Here."] = true
+}
+
+-- Are we in development mode?
+SWAMP_DEV.enabled = false
+-- The delay on checking if we need to refresh files
 SWAMP_DEV.refreshDelay = 2
-SWAMP_DEV.itemStruct = SWAMP_DEV.itemStruct
+
+local currentItemStruct = nil
+
+--- This stores file times for certain root folder.
+--- FileNames[key] -> SaveTime[value]
 SWAMP_DEV.fileTimes = SWAMP_DEV.fileTimes or {}
+ 
+-- lua/STRUCT/*
 
-SWAMP_DEV.structTypes = {
+SWAMP_DEV.structEnvironments = {
     ["weapons"] = function(curClass)
```

Comment:
I just realized this is missing gamemode (gamemode files use GM, not the same as GAMEMODE)